### ISSUE TITLE
Fields that take a number supports scientific notation

### ISSUE THREAD

**Issue by onbjerg** [1Hive/honey-pot#151]
Fields that take a number generally accepts scientific notation (e.g. 1e18 or just e) and other "invalid" inputs such as "-10" and so on.

Areas affected:

- When supporting a proposal initially
- When creating a funding proposal
- When changing support

Low priority bug since the user will have to input this intentionally and transactions will revert in the case that the number is out of range (amount < 0 or amount > balance).

Credit to @anthonyoliai for reporting it on Discord first

**Comment by vylettmonstr** [comment]
Same issue in honeycomb as well

**Comment by famole** [comment]
Hey @onbjerg, 

-Founding proposal, 
 * for negative numbers when you loose the focus, the field is cleaned.
 * for scientifical notation this is what it shows
<img width="442" alt="Screen Shot 2020-10-25 at 11 27 47 AM" src="https://user-images.githubusercontent.com/1489405/97110005-67b8fb80-16b5-11eb-9268-fc9805b260e1.png">

Should we clean the field as negative numbers?

**Comment by onbjerg** [comment]
Yes, it doesn't make sense to have an infinite request amount, the transaction will just revert.

**Comment by kamikazebr** [comment]
It was solved? Need help?

**Comment by kamikazebr** [comment]
I made some tests to undestand, the test below pass. 

![image](https://user-images.githubusercontent.com/691510/146693199-f144e888-232b-4a26-8198-203509a8850a.png)

Its look correct?

So if find some e or minus need be 0 value, right? Test which pass below
![image](https://user-images.githubusercontent.com/691510/146693565-043a7117-2105-4355-8baa-7ad05683b491.png)
